### PR TITLE
Save the actual image when comparing and verifying

### PR DESCRIPTION
**Pull request by takahirom**
https://github.com/takahirom/roborazzi/issues/50

### CODE REVIEW

File: roborazzi/src/main/java/com/github/takahirom/roborazzi/Roborazzi.kt [hunk 1]
```diff
@@ -547,6 +547,12 @@ private fun saveOrCompare(
   goaldenFile: File,
   roborazziOptions: RoborazziOptions
 ) {
+  val forbiddenFileSuffixes = listOf("_compare", "_actual")
+  forbiddenFileSuffixes.forEach {
+    if (goaldenFile.nameWithoutExtension.endsWith(it)) {
```
Comment (yschimke) [inline roborazzi/src/main/java/com/github/takahirom/roborazzi/Roborazzi.kt:552]: nit: 
```suggestion
    if (goldenFile.nameWithoutExtension.endsWith(it)) {
```

**Comment by Bencodes** [comment]
@takahirom are you still planning on merging this one?

**Comment by takahirom** [comment]
@Bencodes 
I haven't merged the change yet because I'm not confident that it's exactly what you're looking for. Could you please confirm if the proposed solution works for you?

**Comment by Bencodes** [comment]
> Could you please confirm if the proposed solution works for you?

The behavior that we are looking for is to be able to record, verify against the existing baseline, and export the new baseline (even if they changed) in a single pass so that we can copy out the updated baseline images when the user decides to make them the new baselines. Having the actuals available alongside the original baselines also makes it easier to post the images directly to Github as a comment for comparison without having to do subsequent builds to get the new snapshots out.

**Comment by takahirom** [comment]
@Bencodes 
Thanks for your feedback. I can add a verifyAndRecordRoborazziDebug task, but there is an issue when a single test contains multiple captureRoboImage() calls and each of them is different. In this case, the first mismatch would cause an assertion, preventing the rest of the images from being updated. This behavior is designed to easily identify and resolve problems using the stack trace of the location where the issue occurred. Is this still acceptable to you?
https://github.com/takahirom/roborazzi/issues/23#issuecomment-1515090642

**Comment by takahirom** [comment]
Even if we need to rerun the task, implementing verifyAndRecordRoborazziDebug might be a good choice. This is because it adheres to the best practice of keeping tests separate. Also, even in the context of a scenario test, this approach can be beneficial because if multiple images are saved, it can become difficult to pinpoint the cause of failure.

**Comment by Bencodes** [comment]
> This behavior is designed to easily identify and resolve problems using the stack trace of the location where the issue occurred. Is this still acceptable to you?

I think this makes sense. And I agree that in terms of best practices one-snapshot-per-test is the correct approach.

**Comment by takahirom** [comment]
Regardless, I'm planning to add the implementation to this PR. So, I'll go ahead and merge this for now.

**Pull request merged**